x-pack/plugins/fleet/public/services/get_cloud_formation_template_url_from_package_info.ts

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { PackageInfo, RegistryPolicyTemplate, RegistryVarsEntry } from '../types';
+
+type RegistryPolicyTemplateWithInputs = RegistryPolicyTemplate & {
+  inputs: Array<{
+    vars?: RegistryVarsEntry[];
+  }>;
+};
+
+// type guard for checking inputs
+export const hasPolicyTemplateInputs = (
+  policyTemplate: RegistryPolicyTemplate
+): policyTemplate is RegistryPolicyTemplateWithInputs => {
+  return policyTemplate.hasOwnProperty('inputs');
+};
+
+/**
+ * Get the cloud formation template url from the PackageInfo
+ * It looks for a input var with a object containing cloud_formation_template_url present in
+ * the package_policies inputs of the given integration type
+ */
+export const getCloudFormationTemplateUrlFromPackageInfo = (
+  packageInfo: PackageInfo | undefined,
+  integrationType: string
+): string | undefined => {
+  if (!packageInfo?.policy_templates) return undefined;
+
+  const policyTemplate = packageInfo.policy_templates.find((p) => p.name === integrationType);
+  if (!policyTemplate) return undefined;
+
+  const inputs = hasPolicyTemplateInputs(policyTemplate) && policyTemplate.inputs;
+
+  if (!inputs) return undefined;
+
+  const cloudFormationTemplate = inputs.reduce((acc, input): string => {
+    if (!input.vars) return acc;
+    const template = input.vars.find((v) => v.name === 'cloud_formation_template')?.default;
+    return template ? String(template) : acc;
+  }, '');
+
+  return cloudFormationTemplate !== '' ? cloudFormationTemplate : undefined;
```

Comment:
We can infer `policyTemplate` directly to discriminate its type to `RegistryPolicyIntegrationTemplate` naturally and remove `hasPolicyTemplateInputs` since its not being used outside of this file

```suggestion
  if ("inputs" in policyTemplate) {
      const cloudFormationTemplate = policyTemplate.inputs?.reduce((acc, input): string => {
        if (!input.vars) return acc;
        const template = input.vars.find((v) => v.name === 'cloud_formation_template')?.default;
        return template ? String(template) : acc;
      }, '');

      return cloudFormationTemplate !== '' ? cloudFormationTemplate : undefined;
    }
```

Reply:
TIL!